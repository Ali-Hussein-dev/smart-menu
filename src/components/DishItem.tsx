import { useState } from 'react'
import { Drawer } from 'antd'
import Price from './Price'
import { ImInfo } from 'react-icons/im'

export const DrawerTitle: React.FC<{ name: string }> = ({ name: cuisine }) => (
  <span className=" text-blueGray-600">
    <div className="flex items-start gap-x-2">
      <ImInfo className="mt-1 text-blueGray-400" />{' '}
      <div className="flex flex-col items-start justify-start">
        <span>Zusatzstoffe Info:</span>
        <span className="text-sm font-thin">{cuisine}</span>
      </div>
    </div>
  </span>
)
interface Props {
  name: string
  description: string
  price: number
}
//=======================
const DishItem: React.FC<Props> = ({ price, name, description }) => {
  // hooks
  const [visible, setVisible] = useState(false)
  //--------------------------------------
  // functions
  //--------------------------------------
  return (
    <div className="flex justify-between bg-blueGray-400">
      <div className="">
        <span className="font-semibold">{name}</span>
        <p className="">{description}</p>
      </div>
      <Price price={price} visible={visible} setVisible={setVisible} />
      <Drawer
        title={<DrawerTitle name={name} />}
        placement={'bottom'}
        closable={true}
        onClose={() => setVisible(false)}
        visible={visible}
        key={'bottom'}
        headerStyle={{
          backgroundColor: '#F8FAFC',
          color: '#334155',
        }}
        bodyStyle={{
          color: '#94A3B8',
        }}
      >
        <p data-testid="drawer">Keine Vorgabe!</p>
      </Drawer>
    </div>
  )
}
export default DishItem
