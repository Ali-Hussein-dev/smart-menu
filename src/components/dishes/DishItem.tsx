import { GetDishes, Price, SvgIcon } from '../leaf-node'
import { useDisclosure } from '@chakra-ui/react'
import {
  Drawer,
  DrawerBody,
  DrawerHeader,
  DrawerOverlay,
  DrawerContent,
} from '@chakra-ui/react'
import { LongText } from '../leaf-node'

interface Props extends Menu.Item, Menu.DishItem {}
//=======================
export const DishItem: React.FC<{ item: Props }> = ({ item }) => {
  // hooks
  const { isOpen, onClose, onToggle } = useDisclosure()
  //--------------------------------------
  // functions
  //--------------------------------------
  return (
    <div className="flex justify-between mb-4 text-blueGray-600">
      <div className="">
        <div className="flex items-center gap-x-1">
          {item.info?.eatStyle?.infoList.filter((o) => o.label === 'Vegan')
            .length === 1 && (
            <SvgIcon size="w-6 h-6">
              <GetDishes iconName="vegan" color="bg-green-500" />
            </SvgIcon>
          )}
          <span className="leading-tighter">{item.name}</span>
        </div>
        <LongText>{item.description}</LongText>
        {item.dressing && (
          <span className="flex items-center gap-x-1 text-blueGray-500">
            <SvgIcon size="w-10 h-10 text-blue-300">
              <GetDishes iconName="dressing" />
            </SvgIcon>
            {item.dressing}
          </span>
        )}
      </div>
      <Price
        isInfo={true}
        price={item.price}
        isOpen={isOpen}
        onToggle={onToggle}
      />
      <Drawer placement={'bottom'} isOpen={isOpen} onClose={onClose} size="md">
        <DrawerOverlay>
          <DrawerContent>
            <DrawerHeader borderBottomWidth="1px">{item.name}</DrawerHeader>
            <DrawerBody>
              <h2 className="font-semibold">{item.info?.allergen?.header}</h2>
            </DrawerBody>
          </DrawerContent>
        </DrawerOverlay>
      </Drawer>
    </div>
  )
}
