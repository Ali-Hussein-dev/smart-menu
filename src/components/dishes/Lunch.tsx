import { withinTimeRange } from 'src/utils'
import dayjs from 'dayjs'
import { Ctx } from 'src/context'
import * as React from 'react'
// import { MdDateRange, MdAccessTime } from 'react-icons/md'
// import { IoPricetagsOutline } from 'react-icons/io5'
//=======================
export const Lunch: React.FC<{ currentTime: number }> = ({ currentTime }) => {
  // hooks
  const { state, dispatch } = React.useContext(Ctx)
  //--------------------------------------
  // functions
  React.useEffect(() => {
    if (
      state.meta.timeRanges.lunch.unavailableDays.filter((num) => {
        num === dayjs().get('d')
      }).length === 1
    ) {
      dispatch({ type: 'offLunch' })
    } else if (
      withinTimeRange(
        currentTime,
        //------------------make lunch available 3hrs earlier
        state.meta.timeRanges.lunch.start - 180,
        state.meta.timeRanges.lunch.end
      )
    ) {
      dispatch({ type: 'onLunch' })
    } else {
      dispatch({ type: 'offLunch' })
    }
  }, [currentTime])
  //--------------------------------------
  switch (state.meta.isLunch) {
    case true:
      return (
        <section className="text-blueGray-600">
          <div className="flex flex-col pl-2 mb-2 gap-y-2">
            <h2 className="font-bold text-center">Mittagessen</h2>
            {/* <span className="flex items-center gap-x-2">
              <MdDateRange size="20" /> 12.3.2021 - 20.3.2021 | Montag bis
              Freitag
            </span>
            <span className="flex items-center gap-x-2">
              <MdAccessTime size="20" />
              11:30-17:00 Uhr
            </span>
            <span className="flex items-center gap-x-2">
              <IoPricetagsOutline size="20" />
              8.80â‚¬
            </span> */}
          </div>
          <div className="px-2">
            <img
              className="w-full mx-auto mb-4 object-fill rounded-lg"
              src="https://www.foodiesfeed.com/wp-content/uploads/2021/01/fried-salmon-with-sweet-soy-sauce-in-a-korean-restaurant.jpg"
            />
          </div>
        </section>
      )
    default:
      return null
  }
}
