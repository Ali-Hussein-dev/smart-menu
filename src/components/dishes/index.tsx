export * from './Lunch'
export * from './DishItem'
