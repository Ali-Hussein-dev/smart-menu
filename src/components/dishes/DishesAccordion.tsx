import {
  Accordion,
  AccordionItem,
  AccordionPanel,
  AccordionButton,
} from '@chakra-ui/react'
import { DishItem } from './'
import { CC, GetDishes, MenuTop, Price } from '../leaf-node'
import * as React from 'react'
import { Lunch } from './Lunch'
import dayjs from 'dayjs'

//=======================
const DishesAccordion: React.FC<{
  dishes: [Menu.Header, Menu.List<Menu.Item>][]
}> = ({ dishes: list }) => {
  // hooks
  //--------------------------------------
  // functions
  //--------------------------------------
  return (
    <>
      <Lunch currentTime={dayjs().get('h') * 60 + dayjs().get('m')} />
      <Accordion allowToggle>
        {list.map((o, i) => (
          <AccordionItem key={i}>
            <AccordionButton
              _focus={{ bg: 'gray.400', outline: 'none' }}
              _expanded={{ bg: 'gray.200', boxShadow: 'md' }}
            >
              <CC isTrue={o[0].iconName}>
                <MenuTop
                  header={o[0].header}
                  Component={<GetDishes iconName={o[0].iconName} />}
                />
              </CC>
            </AccordionButton>
            <AccordionPanel>
              {o[1].list.map((arr, i) => (
                <div key={i}>
                  <CC isTrue={arr[0]?.header}>
                    <h3 className="mb-5 font-semibold text-center">
                      {arr[0].header}
                    </h3>
                  </CC>
                  <MenuTop
                    header={arr[0].header}
                    assest={{ imgSrc: arr[0].imgSrc }}
                  />
                  {arr[1].itemsList.map((oo, ii) => (
                    <DishItem key={ii} item={oo} />
                  ))}
                  <CC isTrue={arr[1].addon}>
                    <div className="addon-list">
                      <h4 className="mb-3 font-semibold">
                        {arr[1].addon?.header}
                      </h4>
                      {arr[1].addon?.addonList.map((ooo, iii) => (
                        <div
                          key={iii}
                          className="flex justify-between mb-2 text-blueGray-600"
                        >
                          <span>{ooo.name}</span>
                          <Price price={ooo.price} />
                        </div>
                      ))}
                    </div>
                  </CC>
                </div>
              ))}
            </AccordionPanel>
          </AccordionItem>
        ))}
      </Accordion>
    </>
  )
}
export default DishesAccordion
