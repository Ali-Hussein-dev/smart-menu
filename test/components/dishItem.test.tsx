import { render } from '@testing-library/react'
import DishItem, { DrawerTitle } from '../../src/components/DishItem'
import dishes from '../../database/dishes/dishes'
describe('DishItem', () => {
  test('DishItem matches snapshot', () => {
    const { container } = render(
      <DishItem
        price={dishes[4].list[0].price}
        name={dishes[4].list[0].name}
        description={dishes[4].list[0].description}
      />
    )
    expect(container).toMatchSnapshot()
  })
  test('DrawerTitle matches snapshot', () => {
    const { container } = render(<DrawerTitle name={dishes[4].list[0].name} />)
    expect(container).toMatchSnapshot()
  })
})
