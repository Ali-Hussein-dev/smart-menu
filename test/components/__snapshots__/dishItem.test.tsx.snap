// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DishItem matches snapshot 1`] = `
<div>
  <div
    class="flex justify-between mb-4 text-blueGray-600"
  >
    <div
      class=""
    >
      <div
        class="flex items-center gap-x-1"
      >
        <div
          class="fill-current w-6 h-6 "
          data-testid="svg-icon"
        >
          <svg
            class="bg-green-500 rounded-full"
            viewBox="0 0 58 58"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
            >
              <path
                d="M29 0v58C12.984 58 0 45.016 0 29S12.984 0 29 0z"
              />
              <g
                fill="#fff"
              >
                <path
                  d="M28.33 32a38.26 38.26 0 00-15.15-17.61 2.08 2.08 0 00-2.64.31 2 2 0 00.43 3c10.2 6.44 16 16.58 16 27.83v3.37a2.07 2.07 0 001.72 2.1A2 2 0 0031 49v-3.45c0-6.62 2.54-13.17 7.56-19.48a2 2 0 00-.59-3 2.07 2.07 0 00-2.6.58A42.71 42.71 0 0030.15 32a1 1 0 01-1.82 0"
                />
                <path
                  d="M33.61 26.06c-2.16-2.35-2.13-6.3.16-9.11C36.18 14 44.91 13 44.91 13s2.84 6.69-.6 10.91c0 0-2.56 3.44-8.93 3.06z"
                />
              </g>
            </g>
          </svg>
        </div>
        <span
          class="leading-tighter"
        >
          Food Item
        </span>
      </div>
      <div
        class="leading-5 break-words cursor-default text-blueGray-400"
        data-testid="text-expand"
      >
        <div
          class="chakra-text css-16ytai3"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Autem similique ipsam debitis nobis alias quasi molestias expedita! Eos, totam nostrum!
        </div>
      </div>
      <span
        class="flex items-center gap-x-1 text-blueGray-500"
      >
        <div
          class="fill-current w-10 h-10 text-blue-300 "
          data-testid="svg-icon"
        >
          <svg
            viewBox="0 0 512.039 512.039"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M175.655 332.952a92.2442 92.2442 0 01-51.2-23.381 91.134 91.134 0 01-29.781-47.701 107.5198 107.5198 0 00-35.072-56.832c-6.315-5.888-12.032-10.325-17.579-14.421-3.794-2.804-9.143-2.002-11.947 1.792-2.804 3.794-2.002 9.143 1.792 11.947 5.12 3.755 10.496 8.192 16.469 13.483a90.455 90.455 0 0129.696 47.701 108.2028 108.2028 0 0035.243 56.832 109.486 109.486 0 0061.099 27.477h.768l-.256.171c4.713.212 8.705-3.436 8.917-8.149.213-4.715-3.436-8.707-8.149-8.919z"
            />
            <path
              d="M399.129 173.791c-3.016-3.621-8.397-4.111-12.018-1.095a136.0134 136.0134 0 01-41.813 23.552c-36.838 12.619-77.055 11.067-112.811-4.352-4.086-1.535-8.673.261-10.63 4.163-2.113 4.213-.41 9.341 3.803 11.453 39.783 17.018 84.487 18.57 125.355 4.352a157.0419 157.0419 0 0032.171-15.616c-7.765 76.8-66.048 136.533-136.533 136.533H177.96c-4.713 0-8.533 3.82-8.533 8.533 0 4.713 3.821 8.533 8.533 8.533h68.693c85.163 0 154.453-76.8 154.453-170.667a8.5267 8.5267 0 00-1.977-5.389z"
            />
            <path
              d="M172.541 332.994c-4.571-1.155-9.213 1.615-10.368 6.187 0 1.195-7.253 27.733-34.133 27.733-4.713 0-8.533 3.82-8.533 8.533 0 4.713 3.82 8.533 8.533 8.533 24.133-.532 44.91-17.182 50.688-40.619 1.154-4.57-1.616-9.212-6.187-10.367zM290.173 366.914c-26.539 0-34.133-26.539-34.133-27.733-1.155-4.713-5.911-7.597-10.624-6.443-4.713 1.155-7.597 5.911-6.443 10.624 5.827 23.618 26.876 40.317 51.2 40.619 4.713 0 8.533-3.82 8.533-8.533 0-4.713-3.821-8.534-8.533-8.534z"
            />
            <path
              d="M392.573 366.914H25.639c-4.713 0-8.533 3.82-8.533 8.533 0 4.713 3.82 8.533 8.533 8.533h366.933c4.713 0 8.533-3.82 8.533-8.533 0-4.713-3.82-8.533-8.532-8.533z"
            />
            <path
              d="M349.906 366.914h-281.6a8.5348 8.5348 0 00-4.206.954c-4.183 2.17-5.815 7.321-3.645 11.504l7.851 15.275c7.618 14.634 22.842 23.716 39.339 23.467h202.923c16.496.25 31.721-8.832 39.339-23.467l7.595-15.275c.631-1.218.959-2.571.956-3.943-.011-4.712-3.839-8.524-8.552-8.515zm-15.445 19.883c-4.652 8.865-13.883 14.37-23.893 14.251H107.645c-10.01.12-19.242-5.386-23.893-14.251l-1.451-2.816h253.611l-1.451 2.816zM449.917 95.725c-37.961-8.353-76.177 12.972-89.003 49.664a50.177 50.177 0 000 33.109c4.441 12.388 16.372 20.492 29.525 20.053 4.713-.471 8.151-4.674 7.68-9.387s-4.674-8.151-9.387-7.68c-5.525-.252-10.3-3.936-11.947-9.216a33.5355 33.5355 0 01.683-21.675c10.017-28.015 39.123-44.387 68.267-38.4 27.392 7.253 49.237 36.267 49.237 66.987-.217 31.13-18.407 59.329-46.677 72.363a163.6642 163.6642 0 01-37.205 9.728c-6.741 1.28-13.739 2.475-20.565 4.181-32.085 7.595-49.152 25.515-49.152 50.261 0 4.713 3.82 8.533 8.533 8.533 4.713 0 8.533-3.82 8.533-8.533 0-20.565 19.627-29.696 36.011-33.792 6.4-1.536 13.056-2.731 19.541-3.925a179.185 179.185 0 0041.131-10.923c34.367-15.833 56.529-50.057 56.917-87.893-.03-38.476-25.272-72.386-62.122-83.455zM232.487 192.152a137.0664 137.0664 0 01-31.915-19.541c-29.424-23.817-66.255-36.586-104.107-36.096a166.1353 166.1353 0 00-87.893 24.661C2.91 164.425-.403 170.62.039 177.133c.353 6.485 4.355 12.207 10.325 14.763a107.371 107.371 0 0121.589 12.885c2.906 1.99 6.736 1.99 9.643 0a76.8096 76.8096 0 0140.96-11.179 75.3482 75.3482 0 0142.667 12.459 80.8029 80.8029 0 0022.528 11.093c3.417.912 5.818 3.974 5.888 7.509v22.784c0 14.138 11.461 25.6 25.6 25.6 1.448.124 2.904.124 4.352 0 12.681-2.496 21.702-13.79 21.333-26.709v-21.845c-.126-3.992 2.533-7.537 6.4-8.533a108.577 108.577 0 0021.76-8.533 8.5281 8.5281 0 003.695-3.996c1.93-4.3.008-9.35-4.292-11.279zm-25.941 7.765c-11.088 3.068-18.768 13.156-18.773 24.661v21.845c.144 4.378-2.76 8.274-6.997 9.387a8.5295 8.5295 0 01-6.997-1.877 8.5336 8.5336 0 01-3.072-6.485v-22.784c-.005-11.025-7.207-20.754-17.749-23.979a63.5787 63.5787 0 01-17.835-8.533 92.5808 92.5808 0 00-52.992-15.616 95.7413 95.7413 0 00-44.8 10.667 117.6571 117.6571 0 00-20.224-11.605 149.1641 149.1641 0 0179.36-22.016c33.864-.483 66.83 10.898 93.184 32.171a148.414 148.414 0 0019.2 13.483l-2.305.681zM185.298 292.589l-1.28-1.024a6.3098 6.3098 0 00-1.536-.768 5.2855 5.2855 0 00-1.621-.512 7.0776 7.0776 0 00-3.243 0 5.277 5.277 0 00-1.621.512c-.545.185-1.061.443-1.536.768l-1.28 1.024a8.5342 8.5342 0 00-2.475 6.059 8.5274 8.5274 0 002.475 5.973 8.5333 8.5333 0 006.059 2.56 8.513 8.513 0 003.243-.683 8.5448 8.5448 0 002.816-1.877 8.5378 8.5378 0 002.475-5.973 8.9692 8.9692 0 00-2.476-6.059z"
            />
          </svg>
        </div>
        Honig-Balsamico
      </span>
    </div>
    <div
      class="flex flex-col items-center justify-between ml-1 text-blueGray-600"
    >
      <div
        class="flex items-center justify-center"
      >
        <span
          class="flex flex-col"
        >
          <span
            class="flex justify-center gap-x-1"
          >
            <span>
              30.00
              <span
                class="pl-1 text-blueGray-500"
              >
                â‚¬
              </span>
            </span>
          </span>
        </span>
      </div>
      <div
        class="flex flex-col items-center w-full"
      >
        <button
          aria-label="item-info"
          class="pb-1"
          role="button"
        >
          <svg
            class="text-blue-300 "
            fill="currentColor"
            height="15"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 16 16"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 4.75c0-0.412 0.338-0.75 0.75-0.75h0.5c0.412 0 0.75 0.338 0.75 0.75v0.5c0 0.412-0.338 0.75-0.75 0.75h-0.5c-0.412 0-0.75-0.338-0.75-0.75v-0.5z"
            />
            <path
              d="M10 12h-4v-1h1v-3h-1v-1h3v4h1z"
            />
            <path
              d="M8 0c-4.418 0-8 3.582-8 8s3.582 8 8 8 8-3.582 8-8-3.582-8-8-8zM8 14.5c-3.59 0-6.5-2.91-6.5-6.5s2.91-6.5 6.5-6.5 6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5z"
            />
          </svg>
        </button>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-1647zlw"
          role="separator"
        />
      </div>
    </div>
  </div>
</div>
`;
